### DISCUSSION TITLE
How to check if 2 SMILES are a pair of enantiomers?

### DISCUSSION THREAD

**Discussion by LiuCMU** [rdkit/rdkit#7169] [FAQ] [answered]
I want to check if a pair of isomers are enantiomers. An examples is below. 

![image](https://github.com/rdkit/rdkit/assets/60156077/481ddef5-24fa-480e-b356-c2d6abaa1684)
```Python
mol4_1 = "C[C@@H](/C=C\C[C@H](Br)C)C(O)=O"
mol4_2 = "C[C@@H](C(O)=O)/C=C\C[C@H](C)Br"
```
My thought is as follows: First find the chiral center indexes and their chiral tags, then comparing the tags of the same atoms. If all chiral centers have opposite chiral tags, then this pair of isomer is a pair of enantiomers. However, the indexes of corresponding atoms in the two isomers are different. For this example, the indexes and chiral tags are `(1, 'S'), (5, 'R')` and `(1, 'R'), (8, 'S')`, respectively.

```python
print(Chem.FindMolChiralCenters(Chem.MolFromSmiles(mol4_1), useLegacyImplementation=False))  # output [(1, 'S'), (5, 'R')]
print(Chem.FindMolChiralCenters(Chem.MolFromSmiles(mol4_2), useLegacyImplementation=False))  # output [(1, 'R'), (8, 'S')]
```
Does anyone have ideas about identifying enantiomers? That would be super helpful. Thanks in advance!

My rdkit version is as follows:
```bash
(py39) tmpry$mamba list rdkit
# packages in environment at /Users/liu5/miniforge3/envs/py39:
#
# Name                    Version                   Build  Channel
rdkit                     2023.03.3        py39h40f44b6_1    conda-forge
```

**Comment by ptosco** [comment, marked as answer]
@LiuCMU Converting both SMILES to their canonical form and then checking that both molecules are chiral and that all their parities are opposite should work.
```
from rdkit import Chem

def are_enantiomers(smi1, smi2):
    mol1 = Chem.MolFromSmiles(smi1)
    assert mol1
    mol2 = Chem.MolFromSmiles(smi2)
    assert mol2
    can_smi1 = Chem.MolToSmiles(mol1)
    can_smi2 = Chem.MolToSmiles(mol2)
    return ("@" in can_smi1 and "@" in can_smi2
        and can_smi1.replace(
            "@@", "__DOUBLE_AT__"
        ).replace("@", "@@").replace(
            "__DOUBLE_AT__", "@"
        ) == can_smi2)

mol4_1 = "C[C@@H](/C=C\C[C@H](Br)C)C(O)=O"
mol4_2 = "C[C@@H](C(O)=O)/C=C\C[C@H](C)Br"

are_enantiomers(mol4_1, mol4_2)
True

are_enantiomers("C[C@@H](/C=C\C[C@H](Br)C)C(O)=O", "C[C@@H](C(O)=O)/C=C\C[C@@H](C)Br")
False

are_enantiomers("CC", "CC")
False
```

**Reply by LiuCMU** [reply]
Thank you so much, @ptosco This is nice and clean!

**Comment by bp-kelley** [comment]
You should probably also check that their non isomeric smiles are the same as well:

```
from rdkit import Chem

smi1 = "[C@H](I)(F)Br"
smi2 = "[C@@H](I)(F)Br"

can1 = Chem.CanonSmiles(smi1, useChiral=False)
can2 = Chem.CanonSmiles(smi2, useChiral=False)

iso1 = Chem.CanonSmiles(smi1)
iso2 = Chem.CanonSmiles(smi2)

if can1 == can2 and iso1 != iso2:
    print("chiral differences")
```

**Reply by LiuCMU** [reply]
Thanks for participating the discussion @bp-kelley. 

The following code from @ptosco already makes sure that `can_smi1` is the same as `can_smi2` after replacing `@` and `@@` with `@@` and `@`, respectively. Why should we still check the non isomeric smiles? Thanks!
`can_smi1.replace("@@", "__DOUBLE_AT__").replace("@", "@@").replace("__DOUBLE_AT__", "@") == can_smi2`
